hedera-mirror-rest/accounts.js

```diff
@@ -297,23 +358,91 @@ const getAccounts = async (req, res) => {
 
   // Parse the filter parameters for account-numbers, balance, and pagination
   const parsedQueryParams = req.query;
-  const [tsQuery, tsParams] = utils.parseTimestampQueryParam(parsedQueryParams, 't.consensus_timestamp');
+  const [transactionTsQuery, transactionTsParams] = utils.parseTimestampQueryParam(
+    parsedQueryParams,
+    't.consensus_timestamp'
+  );
+
+  let paramCount = 0;
   const resultTypeQuery = utils.parseResultParams(req);
   const {query, params, order, limit} = utils.parseLimitAndOrderParams(req);
+  const accountIdParams = [encodedId];
+  const tokenBalanceQuery = {
+    query: `account_id = $${++paramCount}`,
+    params: accountIdParams,
+    limit: tokenBalanceResponseLimit.singleAccount,
+  };
+  const entityAccountQuery = {query: `e.id = $${paramCount}`, params: []};
+  const entityStakeQuery = {query: `(es.id = $${paramCount} OR es.id IS NULL)`, params: []};
+
+  const accountBalanceQuery = {query: '', params: []};
+  if (transactionTsQuery) {
+    if (!utils.isEffectiveTimestamp(filters)) {
+      throw InvalidArgumentError.forRequestValidation({
+        code: InvalidArgumentError.INVALID_PARAM_USAGE,
+        key: filterKeys.TIMESTAMP,
+        error: 'No effective time range',
+      });
+    }
+
+    // Exclude ne query params for balance file
+    const cleanedReqParams = Object.assign({}, parsedQueryParams);
+    const tsParamValue = cleanedReqParams[filterKeys.TIMESTAMP];
+    cleanedReqParams[filterKeys.TIMESTAMP] = (Array.isArray(tsParamValue) ? tsParamValue : [tsParamValue]).filter(
+      (value) => !value.startsWith(constants.queryParamOperators.ne)
+    );
+
+    const [balanceFileTsQuery, balanceFileTsParams] = utils.parseTimestampQueryParam(
+      cleanedReqParams,
+      'consensus_timestamp',
+      {[opsMap.eq]: opsMap.lte}
+    );
+
+    let tokenBalanceTsQuery = transactionTsQuery
+      .replaceAll('t.consensus_timestamp', 'created_timestamp')
+      .replaceAll('?', (_) => `$${++paramCount}`);
+
+    tokenBalanceQuery.query += ` and ${tokenBalanceTsQuery} `;
+    tokenBalanceQuery.params = tokenBalanceQuery.params.concat(transactionTsParams);
+
+    const {conditions: entityTsQuery, params: entityTsParams} = utils.extractTimestampRangeConditionFilters(
+      filters.filter((filter) => filter.key === filterKeys.TIMESTAMP),
+      paramCount
+    );
+    paramCount += entityTsParams.length;
+
+    entityAccountQuery.query += ` and ${entityTsQuery.join(' and ')}`;
+    entityAccountQuery.params = entityAccountQuery.params.concat(entityTsParams);
+
+    const accountBalanceTs = await balances.getAccountBalanceTimestamp(balanceFileTsQuery, balanceFileTsParams, order);
+    const neqTsFilters = filters.filter(
+      (filter) => filter.key === filterKeys.TIMESTAMP && filter.operator === opsMap.ne
+    );
+    const timestampExcluded = neqTsFilters.reduce(
+      (prev, curr) => prev || parseInteger(curr.value) === accountBalanceTs,
```

Comment:
it should be cheaper to just do `==` with auto type coercion, for timestamp, filter.value is a string, `accountBalanceTs` is either a number or a BigInt
 
```suggestion
      (prev, curr) => prev || curr.value == accountBalanceTs,
```